test_framework: python3.4 authproxy compat

laanwj opened the pull request:
Changed in version 3.4: The strict parameter was removed. HTTP 0.9-style “Simple Responses” are not longer supported. (https://docs.python.org/3/library/http.client.html)

Work around this by using a named parameter.

Also avoid an infinite loop in encoding, by ensuring EncodeDecimal returns a string. round(Decimal) used to convert it to float, but it no longer does. Strings are supported since #6380, so just use that.

MarcoFalke commented on qa/rpc-tests/test_framework/authproxy.py:
```diff
@@ -96,7 +96,7 @@ def __init__(self, service_url, service_name=None, timeout=HTTP_TIMEOUT, connect
         else:
             self.__conn = httplib.HTTPConnection(self.__url.hostname, port,
-                                                 False, timeout)
+                                                 timeout=timeout)
```
Thanks!

This is a bugfix, which should also go upstream?

laanwj replied:
Yes, probably.
Although the EncodeDecimal change does break compatibility with older versions of bitcoind.

MarcoFalke replied:
Sure, so I'd prefer those to be two commits :)

MarcoFalke replied:
Actually this was already fixed: https://github.com/jgarzik/python-bitcoinrpc/commit/7ebeebb4f61917fe590d980cb4f9aefdce2c8f25

But somehow we forgot to fetch it.

laanwj replied:
Hm @luke-jr's commit seems to be more thorough. I indeed didn't test the https path. Will replace with his commit.

laanwj committed:
test_framework: Avoid infinite loop in encoding Decimal

Avoid an infinite loop in encoding, by ensuring EncodeDecimal
returns a string. round(Decimal) used to convert it to
float, but it no longer does in python 3.x. Strings are
supported since #6380, so just use that.

laanwj commented:
@MarcoFalke Ok, split into two commits

MarcoFalke commented:
Tested ACK 28b1ec3

sipa commented:
utACK

laanwj commented:
Travis fail was unrelated (#7470), retriggered.

luke-jr committed:
test_framework: Py3.4 compat: Specify timeout parameter by name

Changed in version 3.4: The strict parameter was removed. HTTP 0.9-style
“Simple Responses” are not longer supported.
(https://docs.python.org/3/library/http.client.html)

Source: https://github.com/jgarzik/python-bitcoinrpc/commit/7ebeebb4f61917fe590d980cb4f9aefdce2c8f25

MarcoFalke commented:
utACK e7e48ba
